### PR TITLE
add PyPI auto-publish

### PR DESCRIPTION
**Pull request by ocefpaf**
@mwengren this is the first one on my list. When we get the PyPI credentials sorted out this will become a boilerplate PR to the repos we care about publishing.

@benjwadams can you add my and/or @mwengren as a PyPI maintainer so we can generate the token to make this work?

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      shell: bash
+
+    - name: Install build tools
+      run: |
+        python -m pip install --upgrade pip wheel "setuptools>=41.2" setuptools_scm build twine
+
+      shell: bash
+
+    - name: Build binary wheel
+      run: python -m build --sdist --wheel . --outdir dist
```
Comment (ocefpaf) [inline .github/workflows/publish.yml:32]: Wheel will fail b/c we don't have wheels for `cf_units`. Udunits2 make it hard to create one. I'm working on a solution upstream.

Comment (benjwadams) [reply .github/workflows/publish.yml]: I just recently got `cf_units` built under pip and it was pretty nightmarish to build.  I'll probably update the build instructions very soon.

**Comment by benjwadams** [comment]
@ocefpaf, I sent you invites for maintainer status on some of the compliance checker related projects located on PyPI.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ocefpaf) [reply .github/workflows/publish.yml]: We are getting somewhere in https://github.com/SciTools/cf-units/pull/186 and the wheel should make this easier.

**Comment by ocefpaf** [comment]
> @ocefpaf, I sent you invites for maintainer status on some of the compliance checker related projects located on PyPI.

Thanks! It turns [out I'm locked out of my PyPI account :-(](https://github.com/pypa/pypi-support/issues/1201) but we are not ready to merge this one anyway, not until we have wheels for cf_units.

**Comment by bjlittle** [comment]
@benjwadams and @ocefpaf Good news... you're now unblocked!

Thanks to the awesomeness of @ocefpaf we've now released `cf-unit` 3.0.1.post0, which automagically builds and uploads linux wheels to PyPI :+1:

**Comment by ocefpaf** [comment]
@benjwadams I'm still locked out of my PyPI account. Can you create [the TOKEN](https://pypi.org/help/#apitoken) and add it to the secrets on GH?

**Comment by benjwadams** [comment]
@ocefpaf, I was able to create the token, but I don't think I have the proper permissions scope on GitHub to add new secrets.  I had to have @mwengren change the issue template for me as part of #876, for example.

**Comment by ocefpaf** [comment]
> @ocefpaf, I was able to create the token, but I don't think I have the proper permissions scope on GitHub to add new secrets. I had to have @mwengren change the issue template for me as part of #876, for example.

Thanks! @mwengren I also don't have the proper rights here to do so. Do you think you can help us out? This is the last piece to automate releases.

**Comment by benjwadams** [comment]
@ocefpaf, I've added an environment secret named `PYPI_ACCESS_TOKEN`.  Let me know if this ends up working.

**Comment by ocefpaf** [comment]
@benjwadams it is failing with:

```
Warning:  It looks like you are trying to use an API token to authenticate in the package index and your token value does not start with "pypi-" as it typically should. This may cause an authentication error. Please verify that you have copied your token properly if such an error occurs.
```

Not sure if that is b/c the token is wrong or b/c it is an environment secret instead of a repo secret :-/

**Comment by mwengren** [comment]
@ocefpaf @benjwadams Did you confirm PyPi publishing working properly now?  LMK and I will reset changes I made.

**Comment by ocefpaf** [comment]
It is working as expected. Thanks Micah!

**Pull request merged**